tests/recipes/cpc2/test_evaluate.py

```diff
@@ -0,0 +1,119 @@
+"""Tests for the CPC2 evaluation functions."""
+
+import math
+
+import numpy as np
+import pytest
+
+from recipes.cpc2.baseline.evaluate import (
+    compute_scores,
+    kt_score,
+    ncc_score,
+    rmse_score,
+    std_err,
+)
+
+
+@pytest.mark.parametrize(
+    "x, y, expected", [([1, 2, 3], [1, 2, 3], 0), ([0], [1], 1), ([1, 1, 1], [1], 0)]
+)
+def test_rmse_score_ok(x, y, expected):
+    """Test the function rmse_score valid inputs"""
+    assert rmse_score(np.array(x), np.array(y)) == pytest.approx(expected)
+
+
+@pytest.mark.parametrize(
+    "x, y, expected", [([1, 2, 3], [2, 3], ValueError), ([], [], ValueError)]
+)
+def test_rmse_score_error(x, y, expected):
+    """Test the function rmse_score for invalid inputs"""
+    with pytest.raises(expected):
+        result = rmse_score(np.array(x), np.array(y))
+        if np.isnan(result):
```

Comment:
This suggests to me that the `rmse_score()` should perhaps have a `try: ... except: ...` and raise the `ValueError`. Only noticed because if the `if: ...` here.

Reply:
This is a bit tricksy I think. I kind of agree but raising a value error in rmse_score would not be consistent with numpy behaviour. i.e. Numpy will allow division by zero and returns the result as a NaN. So something like RMSE score will return NaNs for empty arrays. I guess numpy users want it to be able to handle missing data and expect NaN results. So this case really belongs in the test_rmse_score_ok as NaN as a valid result, BUT...  you can't have a NaN as an expected results because NaN == NaN is false!  And I didn't want to make a 3rd version of the test for this very specific case, so raising the error on .isnan to make it fit the semantics of this test.